core/object/object.h

```diff
@@ -153,15 +199,29 @@ struct PropertyInfo {
 
 	// If you are thinking about adding another member to this class, ask the maintainer (Juan) first.
 
+	STRUCT_LAYOUT(PropertyInfo,
+			STRUCT_MEMBER(name, Variant::STRING, String()),
+			STRUCT_MEMBER(class_name, Variant::STRING_NAME, StringName()),
+			STRUCT_MEMBER(type, Variant::INT, Variant::NIL),
+			STRUCT_MEMBER(hint, Variant::INT, PROPERTY_HINT_NONE),
+			STRUCT_MEMBER(hint_string, Variant::STRING, String()),
+			STRUCT_MEMBER(usage, Variant::INT, PROPERTY_USAGE_DEFAULT))
+
 	_FORCE_INLINE_ PropertyInfo added_usage(uint32_t p_fl) const {
 		PropertyInfo pi = *this;
 		pi.usage |= p_fl;
 		return pi;
 	}
 
+	operator Struct<PropertyInfo>() const;
+
+	static PropertyInfo from_struct(const Struct<PropertyInfo> &p_struct);
+
+#ifndef DISABLE_DEPRECATED
 	operator Dictionary() const;
 
 	static PropertyInfo from_dict(const Dictionary &p_dict);
```

Comment:
Are you sure these can be removed when deprecated stuff is disabled? These are used in `script_language_extension.h` in methods that are definitely not deprecated. But I haven't looked through the whole PR to see if you've worked around that somewhere.